modules/slate-sugar/src/main/ts/ephox/slate-sugar/api/Api.ts

```diff
@@ -0,0 +1,69 @@
+import { Maybe } from '@ephox/katamari';
+
+type path = number[];
+
+interface Editor {}
+
+interface RtcAdaptor {
+  raw: {
+    getModel: () => Maybe<any>;
+    getModelApi: () => Maybe<ModelApi>;
+  };
+}
+
+interface RtcEditor extends Editor {
+  rtcInstance: RtcAdaptor;
+}
+
+interface ModelElement {
+  type: string;
+  children: Array<ModelNode>;
+  class?: string;
+}
+
+interface ModelText {
+  text: string;
+}
+
+type ModelNode = ModelElement | ModelText;
+
+interface ModelLocation {
+  node: ModelNode;
+  parents: Array<ModelLocation>;
+}
+
+interface ModelApi {
+  body: {
+    getBody: () => ModelLocation;
+  };
+  modelNodeType: {
+    isElement: (node: ModelNode) => node is ModelElement;
+    isText: (node: ModelNode) => node is ModelText;
+    isBlock: (node: ModelNode) => boolean;
+    isInline: (node: ModelNode) => boolean;
+  };
+  nodeTransforms: {
+    setPropsAtPath: (path: path, props: Record<string, string>) => void;
+  };
+  predicateFind: {
+    closest: (loc: ModelLocation, predicate: (loc: ModelLocation) => boolean, isRoot: (loc: ModelLocation) => boolean) => ModelLocation;
+  };
+  predicateFilter: {
+    descendants: (loc: ModelLocation, predicate: (loc: ModelLocation) => boolean) => ModelLocation[];
+  };
+  slateLoc: {
```

Comment:
I think this needs to be renamed to modelLocation as well. Or be exposed in a new path module as modelLocationToPath we will probably expose more path utility functions from the slate apis since they are also handy.

Reply:
done

Reply:
I think I was a bit unclear it could be in a Path module as a function called moduleLocationToPath since we are likely to add more path utility functions later like Path.equals etc.

Reply:
done